apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -564,6 +564,37 @@ type Storage struct {
 // +kubebuilder:validation:Maximum=4094
 type VLANID int32
 
+type LLDPProcessed struct {
+	// https://github.com/openstack/python-ironic-inspector-client/blob/master/ironic_inspector_client/resource.py#L24-L59
+	Interface                         string           `json:"interface,omitempty"`
+	Mac                               string           `json:"mac,omitempty"`
+	NodeIdent                         string           `json:"node_ident,omitempty"`
+	SwitchCapabilitiesEnabled         []string         `json:"switch_capabilities_enabled,omitempty"`
+	SwitchCapabilitiesSupport         []string         `json:"switch_capabilities_support,omitempty"`
+	SwitchChassisId                   string           `json:"switch_chassis_id,omitempty"`
+	SwitchMgmtAddresses               []string         `json:"switch_mgmt_addresses,omitempty"`
+	SwitchPortAutonegotiation_enabled string           `json:"switch_port_autonegotiation_enabled,omitempty"`
+	SwitchPortAutonegotiation_support string           `json:"switch_port_autonegotiation_support,omitempty"`
+	SwitchPortDescription             string           `json:"switch_port_description,omitempty"`
+	SwitchPortId                      string           `json:"switch_port_id,omitempty"`
+	SwitchPortLinkAggregationEnabled  bool             `json:"switch_port_link_aggregation_enabled,omitempty"`
+	SwitchPortLinkAggregationId       int              `json:"switch_port_link_aggregation_id,omitempty"`
+	SwitchPortLinkAggregationSupport  bool             `json:"switch_port_link_aggregation_support,omitempty"`
+	SwitchPortManagementVlanId        string           `json:"switch_port_management_vlan_id,omitempty"`
+	SwitchPortMauType                 string           `json:"switch_port_mau_type,omitempty"`
+	SwitchPortMTU                     int              `json:"switch_port_mtu,omitempty"`
+	SwitchPortPhysicalCapabilities    string           `json:"switch_port_physical_capabilities,omitempty"`
+	SwitchPortProtocolVlanEnabled     string           `json:"switch_port_protocol_vlan_enabled,omitempty"`
+	SwitchPortProtocolVlanIds         string           `json:"switch_port_protocol_vlan_ids,omitempty"`
```

Comment:
It's weird to have VlanIds as `string` in one place and as `[]string` (why not int?) in another. I don't remember why it was done initially, but I suggest we clean it up.